File: app/src/main/java/com/sendbird/calls/quickstart/main/ViewPagerWithSwipeOption.java
package com.sendbird.calls.quickstart.main;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.viewpager.widget.ViewPager;

public class ViewPagerWithSwipeOption extends ViewPager {

    private boolean mIsSwipeEnabled = true;

    public ViewPagerWithSwipeOption(@NonNull Context context) {
        super(context);
    }

    public ViewPagerWithSwipeOption(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        if (mIsSwipeEnabled) {
            return super.onInterceptTouchEvent(ev);
        }
        return false;
    }

    @SuppressLint("ClickableViewAccessibility")
    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        if (mIsSwipeEnabled) {
            return super.onTouchEvent(ev);
        }
        return false;
    }

    @Override
    public void setCurrentItem(int item, boolean smoothScroll) {
        super.setCurrentItem(item, false);
    }

    @Override
    public void setCurrentItem(int item) {
        super.setCurrentItem(item, false);
    }
}

File: directcall-kotlin/app/src/main/java/com/sendbird/calls/quickstart/main/ViewPagerWithSwipeOption.kt
package com.sendbird.calls.quickstart.main

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.viewpager.widget.ViewPager

class ViewPagerWithSwipeOption : ViewPager {
    private val mIsSwipeEnabled = true

    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        return if (mIsSwipeEnabled)  super.onInterceptTouchEvent(ev) else false
    }

    @SuppressLint("ClickableViewAccessibility")
    override fun onTouchEvent(ev: MotionEvent): Boolean {
        return if (mIsSwipeEnabled) super.onTouchEvent(ev) else false
    }

    override fun setCurrentItem(item: Int, smoothScroll: Boolean) {
        super.setCurrentItem(item, false)
    }

    override fun setCurrentItem(item: Int) {
        super.setCurrentItem(item, false)
    }
}